test/browser/spec/ol/renderer/webgl/PointsLayer.test.js

```diff
@@ -688,4 +692,71 @@ describe('ol/renderer/webgl/PointsLayer', function () {
       renderer.renderFrame(frameState);
     });
   });
+
+  describe('#updateStyleVariables()', function () {
+    const targetContext = createCanvasContext2D(1, 1);
+    function getCenterPixelImageData() {
+      targetContext.clearRect(0, 0, 1, 1);
+      const canvas = document.querySelector('.testlayer');
+      targetContext.drawImage(canvas, 50, 50, 1, 1, 0, 0, 1, 1);
+      return targetContext.getImageData(0, 0, 1, 1).data;
+    }
+    let map, layer;
+    beforeEach(function (done) {
+      layer = new WebGLPointsLayer({
+        className: 'testlayer',
+        source: new VectorSource({
+          features: [new Feature(new Point([0, 0]))],
+        }),
+        style: {
+          variables: {
+            r: 0,
+            g: 255,
+            b: 0,
+          },
+          symbol: {
+            symbolType: 'circle',
+            size: 14,
+            color: ['color', ['var', 'r'], ['var', 'g'], ['var', 'b']],
+          },
+        },
+      });
+      map = new Map({
+        pixelRatio: 1,
+        target: createMapDiv(100, 100),
+        layers: [layer],
+        view: new View({
+          center: [0, 0],
+          zoom: 2,
+        }),
+      });
+      map.on('rendercomplete', function onrendercomplete() {
+        //FIXME rendercomplete often fires twiece before we get the expected image data. Why?
```

Comment:
@jahow This is the `rendercomplete` event that will be triggered twice before the layer is rendered as expected, instead of just once.